gcs/COOPERATIVE_LOCKING.md

```diff
@@ -0,0 +1,180 @@
+### Introducing Cloud Storage Connector Cooperative Locking
+
+We are excited to release Cooperative Locking support for Cloud Storage
+Connector 2.0. With this release you will be able to isolate and recover
+directory modification operations performed through `hadoop fs` command line
+interface on Cloud Storage.
+
+The key motivation behind this feature is a fact that directory mutations are
+not atomic on Cloud Storage, which means that competing directory mutations are
+susceptible to race conditions that can lead to data integrity issues (loss,
+duplication, etc).
+
+### How it works
+
```

Comment:
Mention that the scope of lock is per bucket, dir modification operations in the same bucket will be sequential.

Reply:
Done.